templates/terraform/pre_update/deployment_manager_deployment.go.erb

```diff
@@ -0,0 +1,46 @@
+<%-# The license inside this block applies to this file:
+   # Copyright 2018 Google Inc.
+   # Licensed under the Apache License, Version 2.0 (the "License");
+   # you may not use this file except in compliance with the License.
+   # You may obtain a copy of the License at
+   #
+   #     http://www.apache.org/licenses/LICENSE-2.0
+   #
+   # Unless required by applicable law or agreed to in writing, software
+   # distributed under the License is distributed on an "AS IS" BASIS,
+   # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   # See the License for the specific language governing permissions and
+   # limitations under the License.
+-%>
+d.Partial(true)
+if d.HasChange("preview") {
```

Comment:
We can do this with custom updates! https://github.com/GoogleCloudPlatform/magic-modules/blob/master/api/type.rb#L41-L46 specifically. See https://github.com/GoogleCloudPlatform/magic-modules/pull/2475/files. If specific ordering is important, you can probably sort the ids so preview comes first.

This will make the fingerprint stuff automatic too if you use `fingerprint_name`.

Reply:
i would also like to change this? i didn't like that I had to mark the object as input and then add custom update_verb/id/url/fingerprint_name to all the other updatable fields

mostly this is a note for later me

Reply:
Yeah, the UX is a little sad. I think we can fix it better in MMv2, where all updates have their own config & custom updates are less exceptional.

Reply:
👍